{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8ebffed3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Importing necessary libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7bef6654",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/maddi/Documents/secure_genomics_pipeline/notebooks'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7dc1ab01",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load dataset\n",
    "DATA_PATH = \"../data/raw/GSE260586_FPKM+1.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "187cd3cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataset shape: (25384, 19)\n",
      "  Chromosome      Start       Stop  Length Strand      Gene Symbol  \\\n",
      "0          X  100627108  100639992   12884      -  ENSG00000000003   \n",
      "1         20   50934867   50959141   24274      -  ENSG00000000419   \n",
      "2          1  169849631  169894268   44637      -  ENSG00000000457   \n",
      "3          1  169662007  169854081  192074      +  ENSG00000000460   \n",
      "4          1  196651754  196752477  100723      +  ENSG00000000971   \n",
      "\n",
      "     gene_biotype          gene_id gene_name     gene_source  gene_version  \\\n",
      "0  protein_coding  ENSG00000000003    TSPAN6  ensembl_havana            16   \n",
      "1  protein_coding  ENSG00000000419      DPM1  ensembl_havana            14   \n",
      "2  protein_coding  ENSG00000000457     SCYL3  ensembl_havana            14   \n",
      "3  protein_coding  ENSG00000000460  C1orf112  ensembl_havana            17   \n",
      "4  protein_coding  ENSG00000000971       CFH  ensembl_havana            17   \n",
      "\n",
      "       r542     r543     r544     r545      r546      r547     r548     r549  \n",
      "0  11.77530  9.60226  2.11983  2.19268  12.41310  13.80670  6.83946  8.45341  \n",
      "1   5.68263  5.63097  4.95180  3.90354   7.11210   6.30274  3.85806  5.09349  \n",
      "2   1.33695  1.42742  1.20361  1.16155   1.34071   1.33318  1.15681  1.21611  \n",
      "3   1.34010  1.30341  1.15043  1.15294   1.34545   1.34649  1.18074  1.22534  \n",
      "4   1.00001  1.00307  1.11688  1.10830   1.00898   1.00544  1.10518  1.16090  \n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(DATA_PATH, sep=\"\\t\")\n",
    "print(\"Dataset shape:\", df.shape)\n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "457bd7c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After setting GeneID as index:\n",
      "                Chromosome      Start       Stop  Length Strand  \\\n",
      "gene_id                                                           \n",
      "ENSG00000000003          X  100627108  100639992   12884      -   \n",
      "ENSG00000000419         20   50934867   50959141   24274      -   \n",
      "ENSG00000000457          1  169849631  169894268   44637      -   \n",
      "ENSG00000000460          1  169662007  169854081  192074      +   \n",
      "ENSG00000000971          1  196651754  196752477  100723      +   \n",
      "\n",
      "                     Gene Symbol    gene_biotype gene_name     gene_source  \\\n",
      "gene_id                                                                      \n",
      "ENSG00000000003  ENSG00000000003  protein_coding    TSPAN6  ensembl_havana   \n",
      "ENSG00000000419  ENSG00000000419  protein_coding      DPM1  ensembl_havana   \n",
      "ENSG00000000457  ENSG00000000457  protein_coding     SCYL3  ensembl_havana   \n",
      "ENSG00000000460  ENSG00000000460  protein_coding  C1orf112  ensembl_havana   \n",
      "ENSG00000000971  ENSG00000000971  protein_coding       CFH  ensembl_havana   \n",
      "\n",
      "                 gene_version      r542     r543     r544     r545      r546  \\\n",
      "gene_id                                                                        \n",
      "ENSG00000000003            16  11.77530  9.60226  2.11983  2.19268  12.41310   \n",
      "ENSG00000000419            14   5.68263  5.63097  4.95180  3.90354   7.11210   \n",
      "ENSG00000000457            14   1.33695  1.42742  1.20361  1.16155   1.34071   \n",
      "ENSG00000000460            17   1.34010  1.30341  1.15043  1.15294   1.34545   \n",
      "ENSG00000000971            17   1.00001  1.00307  1.11688  1.10830   1.00898   \n",
      "\n",
      "                     r547     r548     r549  \n",
      "gene_id                                      \n",
      "ENSG00000000003  13.80670  6.83946  8.45341  \n",
      "ENSG00000000419   6.30274  3.85806  5.09349  \n",
      "ENSG00000000457   1.33318  1.15681  1.21611  \n",
      "ENSG00000000460   1.34649  1.18074  1.22534  \n",
      "ENSG00000000971   1.00544  1.10518  1.16090  \n",
      "                     r542     r543     r544     r545      r546      r547  \\\n",
      "gene_id                                                                    \n",
      "ENSG00000000003  11.77530  9.60226  2.11983  2.19268  12.41310  13.80670   \n",
      "ENSG00000000419   5.68263  5.63097  4.95180  3.90354   7.11210   6.30274   \n",
      "ENSG00000000457   1.33695  1.42742  1.20361  1.16155   1.34071   1.33318   \n",
      "ENSG00000000460   1.34010  1.30341  1.15043  1.15294   1.34545   1.34649   \n",
      "ENSG00000000971   1.00001  1.00307  1.11688  1.10830   1.00898   1.00544   \n",
      "\n",
      "                    r548     r549  \n",
      "gene_id                            \n",
      "ENSG00000000003  6.83946  8.45341  \n",
      "ENSG00000000419  3.85806  5.09349  \n",
      "ENSG00000000457  1.15681  1.21611  \n",
      "ENSG00000000460  1.18074  1.22534  \n",
      "ENSG00000000971  1.10518  1.16090  \n",
      "Transposed DataFrame shape: (8, 25384)\n",
      "gene_id  ENSG00000000003  ENSG00000000419  ENSG00000000457  ENSG00000000460  \\\n",
      "r542            11.77530          5.68263          1.33695          1.34010   \n",
      "r543             9.60226          5.63097          1.42742          1.30341   \n",
      "r544             2.11983          4.95180          1.20361          1.15043   \n",
      "r545             2.19268          3.90354          1.16155          1.15294   \n",
      "r546            12.41310          7.11210          1.34071          1.34545   \n",
      "\n",
      "gene_id  ENSG00000000971  ENSG00000001036  ENSG00000001084  ENSG00000001167  \\\n",
      "r542             1.00001          5.27283          1.60928          2.91309   \n",
      "r543             1.00307          4.78711          2.33586          2.94124   \n",
      "r544             1.11688          5.14611          1.28004          2.25724   \n",
      "r545             1.10830          4.72917          1.29931          2.37244   \n",
      "r546             1.00898          3.09365          2.33214          3.21552   \n",
      "\n",
      "gene_id  ENSG00000001460  ENSG00000001461  ...  ENSG00000292348  \\\n",
      "r542             1.11052          1.65227  ...          1.05009   \n",
      "r543             1.09796          1.46360  ...          1.05514   \n",
      "r544             1.11475          1.45676  ...          1.10137   \n",
      "r545             1.10353          1.46483  ...          1.08730   \n",
      "r546             1.06500          1.80820  ...          1.05258   \n",
      "\n",
      "gene_id  ENSG00000292357  ENSG00000292358  ENSG00000292360  ENSG00000292362  \\\n",
      "r542             1.00145          1.26074          1.00350          1.01555   \n",
      "r543             1.00248          1.20094          1.00557          1.02424   \n",
      "r544             1.09949          1.24102          1.00150          1.02164   \n",
      "r545             1.07708          1.24309          1.00068          1.02686   \n",
      "r546             1.08681          1.19074          1.01294          1.06297   \n",
      "\n",
      "gene_id  ENSG00000292363  ENSG00000292364  ENSG00000292366  ENSG00000292372  \\\n",
      "r542             1.00272          1.80427          1.55816          1.57968   \n",
      "r543             1.00927          1.78662          1.55030          1.59491   \n",
      "r544             1.00640          2.00411          1.79442          1.36413   \n",
      "r545             1.00066          1.79651          1.74271          1.31458   \n",
      "r546             1.00064          2.55384          2.85370          1.47785   \n",
      "\n",
      "gene_id  ENSG00000292373  \n",
      "r542             1.02651  \n",
      "r543             1.01962  \n",
      "r544             1.00493  \n",
      "r545             1.00781  \n",
      "r546             1.01419  \n",
      "\n",
      "[5 rows x 25384 columns]\n"
     ]
    }
   ],
   "source": [
    "#Column 7 is GeneID\n",
    "gene_column = df.columns[7]\n",
    "\n",
    "#Set GeneID as index\n",
    "df = df.set_index(gene_column)\n",
    "\n",
    "print(\"After setting GeneID as index:\")\n",
    "print(df.head())\n",
    "\n",
    "#Drop metadata rows\n",
    "metadata_rows = ['Chromosome', 'Start', 'Stop', 'Length', 'Strand', 'Gene Symbol', 'gene_biotype', 'gene_name', 'gene_source', 'gene_version']\n",
    "df = df.drop(columns=metadata_rows)\n",
    "\n",
    "print(df.head())\n",
    "\n",
    "#Transpose so rows are samples and columns are genes\n",
    "df_t = df.T\n",
    "\n",
    "print(\"Transposed DataFrame shape:\", df_t.shape)\n",
    "print(df.T.head())\n",
    "\n",
    "#Scale the data\n",
    "scaler = StandardScaler()\n",
    "df_scaled = scaler.fit_transform(df_t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44798db1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
